synapse/handlers/room_member.py

```diff
@@ -718,6 +725,30 @@ async def update_membership_locked(
                     # so don't really fit into the general auth process.
                     raise AuthError(403, "Guest access not allowed")
 
+            # Figure out whether the user is a server admin to determine whether they
+            # should be able to bypass the spam checker.
+            if (
+                self._server_notices_mxid is not None
+                and requester.user.to_string() == self._server_notices_mxid
+            ):
+                # allow the server notices mxid to join rooms
+                is_requester_admin = True
```

Comment:
```suggestion
                should_check_join_with_spam_checker = True
```

Or something, as its a bit confusing for `is_requester_admin` to be true when the requester isn't an admin.